### PR TITLE
[Impeller] Fix Gaussian blur jittering.

### PR DESCRIPTION
**Pull request by dnfield**
Converting the scaled texutre width to an ISize snaps it down. This accounts
for the extra snapping in the scale transform.

Also fixes a bug where small blur radius values which would result in a
single turn of the shader sampling would produce no output.

Updates an existing test to use smaller sigma range to make jitter
more apparent on changing the imgui slider.

### CODE REVIEW

**Review by bdero — APPROVED** [review]

**Comment by zanderso** [comment]
Landing this to help it get rolled into the framework.

File: impeller/entity/contents/filters/gaussian_blur_filter_contents.cc [hunk 1]
```diff
@@ -216,11 +222,11 @@ std::optional<Snapshot> DirectionalGaussianBlurFilterContents::RenderFilter(
   };
 
   Scalar x_scale =
-      std::min(1.0, 1.0 / std::ceil(std::log2(transformed_blur_radius_length)));
-  auto out_texture =
-      renderer.MakeSubpass(ISize(pass_texture_rect.size.width * x_scale,
-                                 pass_texture_rect.size.height),
-                           callback);
+      1.0 /
+      std::ceil(std::log2(std::max(2.0f, transformed_blur_radius_length)));
```
Comment (flar) [inline impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:226]: Looking at this line, I'm missing the difference from the old code. Is it impacting the solution to the jitter?

Comment (dnfield) [reply impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:226]: The old code could result in -inf. This code won't.

Comment (flar) [reply impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:226]: D'oh! Yep, I see it...

**Pull request merged**